File: PRACTICA 4/AreaCalculator.py
import math

class Figura:
    def area(self):
        raise NotImplementedError("Este método debe ser implementado por subclases")

class Circulo(Figura):
    def __init__(self, radio):
        self.radio = radio

    def area(self):
        return math.pi * self.radio ** 2

class Rectangulo(Figura):
    def __init__(self, base, altura):
        self.base = base
        self.altura = altura

    def area(self):
        return self.base * self.altura

class Triangulo(Figura):
    def __init__(self, base, altura):
        self.base = base
        self.altura = altura

    def area(self):
        return (self.base * self.altura) / 2

class Trapecio(Figura):
    def __init__(self, base_menor, base_mayor, altura):
        self.base_menor = base_menor
        self.base_mayor = base_mayor
        self.altura = altura

    def area(self):
        return ((self.base_menor + self.base_mayor) * self.altura) / 2

class Pentagono(Figura):
    def __init__(self, lado):
        self.lado = lado

    def area(self):
        return (5 * self.lado ** 2) / (4 * math.tan(math.pi / 5))

# Ejemplo de uso
figuras = [
    Circulo(5),
    Rectangulo(4, 6),
    Triangulo(4, 3),
    Trapecio(3, 5, 4),
    Pentagono(3)
]

for figura in figuras:
    print(f"Área de la figura: {figura.area()}")

File: PRACTICA 4/AreaCalculator.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompany.areacalculator;

/**
 *
 * @author Lenovo
 */
public class AreaCalculator {

    public double area(double radio) {
        return Math.PI * radio * radio; // Círculo
    }

    public double area(double base, double altura) {
        return base * altura; // Rectángulo
    }

    public double area(double base, double altura, boolean esTriangulo) {
        if (esTriangulo) {
            return (base * altura) / 2; // Triángulo
        }
        return 0;
    }

    public double area(double baseMenor, double baseMayor, double altura, String tipo) {
        if (tipo.equals("trapecio")) {
            return ((baseMenor + baseMayor) * altura) / 2; // Trapecio
        }
        return 0;
    }

    public double areaPentagono(double lado) {
        return (5 * lado * lado) / (4 * Math.tan(Math.PI / 5)); // Pentágono
    }

    public static void main(String[] args) {
        AreaCalculator calc = new AreaCalculator();
        System.out.println("Área del círculo: " + calc.area(5));
        System.out.println("Área del rectángulo: " + calc.area(4, 6));
        System.out.println("Área del triángulo: " + calc.area(4, 3, true));
        System.out.println("Área del trapecio: " + calc.area(3, 5, 4, "trapecio"));
        System.out.println("Área del pentágono: " + calc.areaPentagono(3));
    }
}